src/process/personality.rs

```diff
@@ -0,0 +1,1747 @@
+        const NONRECOV = 1;
+        /// Async recoverable exception mode.
+        const ASYNC = 2;
+        /// Precise exception mode.
+        const PRECISE = 3;
+
+        /// Use FPEXC for floating point exception enables.
+        const SW_ENABLE = 0x80;
+        /// Floating point divide by zero.
+        const DIV = 0x01_0000;
+        /// Floating point overflow.
+        const OVF = 0x02_0000;
+        /// Floating point underflow.
+        const UND = 0x04_0000;
+        /// Floating point inexact result.
+        const RES = 0x08_0000;
+        /// Floating point invalid operation.
+        const INV = 0x10_0000;
+    }
+}
+
+/// Get floating point exception mode.
+#[inline]
+pub fn floating_point_exception_mode() -> io::Result<Option<FloatingPointExceptionMode>> {
+    prctl_get_at_arg2_optional::<c_uint>(PR_GET_FPEXC, 0).map(FloatingPointExceptionMode::from_bits)
+}
+
+/// Set floating point exception mode.
+pub const PR_SET_FPEXC: i32 = 12;
+
+/// Set floating point exception mode.
+#[inline]
+pub fn set_floating_point_exception_mode(
+    config: Option<FloatingPointExceptionMode>,
+) -> io::Result<()> {
+    let config = config
+        .as_ref()
+        .map_or(0_u32, FloatingPointExceptionMode::bits);
+    syscall_prctl(PR_SET_FPEXC, config as usize, 0, 0, 0).map(|_r| ())
+}
+
+/*
+ * PR_GET_TIMING/PR_SET_TIMING
+ */
+
+/// Get which process timing method is currently in use.
+pub const PR_GET_TIMING: i32 = 13;
+
+const PR_TIMING_STATISTICAL: i32 = 0;
+const PR_TIMING_TIMESTAMP: i32 = 1;
+
+/// `PR_TIMING_*`.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[repr(i32)]
+pub enum ProcessTimingMethod {
+    /// Normal, traditional, statistical process timing.
+    Statistical = PR_TIMING_STATISTICAL,
+    /// Accurate timestamp based process timing.
+    TimeStamp = PR_TIMING_TIMESTAMP,
+}
+
+impl TryFrom<i32> for ProcessTimingMethod {
+    type Error = io::Errno;
+
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        match value {
+            PR_TIMING_STATISTICAL => Ok(Self::Statistical),
+            PR_TIMING_TIMESTAMP => Ok(Self::TimeStamp),
+            _ => Err(io::Errno::RANGE),
+        }
+    }
+}
+
+/// Get which process timing method is currently in use.
+#[inline]
+pub fn process_timing_method() -> io::Result<ProcessTimingMethod> {
+    syscall_prctl(PR_GET_TIMING, 0, 0, 0, 0)?.try_into()
+}
+
+/// Set whether to use (normal, traditional) statistical process timing or accurate
+/// timestamp-based process timing.
+pub const PR_SET_TIMING: i32 = 14;
+
+/// Set whether to use (normal, traditional) statistical process timing or accurate
+/// timestamp-based process timing.
+#[inline]
+pub fn set_process_timing_method(method: ProcessTimingMethod) -> io::Result<()> {
+    syscall_prctl(PR_SET_TIMING, method as usize, 0, 0, 0).map(|_r| ())
+}
+
+/*
+ * PR_GET_NAME/PR_SET_NAME
+ */
+
+/// Get the name of the calling thread.
+pub const PR_GET_NAME: i32 = 16;
```

Comment:
Would it be feasible to convert all the opcode constants into a Rust `repr(i32)` enum?

Also, if the generic `prctl` is changed to be non-`pub`, these opcodes could be non-`pub` too.